internal/indexer/watched_accounts_cache.go

```diff
@@ -0,0 +1,99 @@
+package indexer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/bits-and-blooms/bloom/v3"
+	"github.com/stellar/go/support/log"
+)
+
+type ReloadFn func(ctx context.Context) ([]string, error)
+
+type WatchedAccountsCache struct {
+	bloomFilter    atomic.Value // holds *bloom.BloomFilter
+	reloadInterval time.Duration
+	reloadFn       ReloadFn
+	quit           chan struct{}
+	closeOnce      sync.Once
+}
+
+func NewWatchedAccountsCache(ctx context.Context, reloadInterval time.Duration, reloadFn ReloadFn) *WatchedAccountsCache {
+	wc := &WatchedAccountsCache{
+		reloadInterval: reloadInterval,
+		reloadFn:       reloadFn,
+		quit:           make(chan struct{}),
+	}
+	go wc.periodicRebuild(ctx)
+	return wc
+}
+
+// periodicRebuild is a goroutine that periodically reloads the watched accounts and updates the bloom filter.
```

Comment:
Why do we need to rebuild the filter by querying the DB? I figured the filter would be updated using the following flow:

- on startup, get all accounts from the DB and build the filter
- when a new account is registered, immediately add it to the bloom filter and insert it into the DB
- when an account is deregistered, immediately remove it from the filter and delete it from the DB

Rebuilding from the DB implies that accounts are getting added to the DB without getting added to the filter at the same time.

Reply:
I can see how if the ingestor and API were running within different processes or machines, we would need some kind of shared data storage like postgres or redis, but I think we should try to support running both the ingestor and API (or other services we build in the future, like webhooks) from within the same process first. We'll probably need to support splitting them into different processes eventually for scalability purposes, but maybe we can keep it as one process for now?

We should be able to do this by running these services in different goroutines and then using channels to send messages (ex. when a new account is registered).